vulnerabilities/importers/gitlab.py

```diff
@@ -0,0 +1,347 @@
+        name = "/".join(parts[1:])
+        return PackageURL(type=purl_type, namespace=None, name=name)
+    # if package slug is of the form:
+    # "nuget/NuGet.Core"
+    if len(parts) == 2:
+        name = parts[1]
+        return PackageURL(type=purl_type, name=name)
+    # if package slug is of the form:
+    # "nuget/github/user/abc/NuGet.Core"
+    if len(parts) >= 3:
+        gitlab_scheme = parts[0]
+        name = parts[-1]
+        namespace = "/".join(parts[1:-1])
+        return PackageURL(type=purl_type, namespace=namespace, name=name)
+    logger.error(f"get_purl: package_slug can not be parsed: {package_slug!r}")
+    return None
+
+
+def extract_affected_packages(
+    affected_version_range: VersionRange,
+    fixed_versions: List[Version],
+    purl: PackageURL,
+) -> Iterable[AffectedPackage]:
+    """
+    Yield a list of AffectedPackage objects
+    """
+    for fixed_version in fixed_versions:
+        yield AffectedPackage(
+            package=purl,
+            fixed_version=fixed_version,
+            affected_version_range=affected_version_range,
+        )
+
+
+def parse_yaml_file(file):
+    """
+    Take file name as input and parse the yaml file
+    to get AdvisoryData object
+
+    Sample YAML file:
+    ---
+    identifier: "GMS-2018-26"
+    package_slug: "packagist/amphp/http"
+    title: "Incorrect header injection check"
+    description: "amphp/http isn't properly protected against HTTP header injection."
+    pubdate: "2018-03-15"
+    affected_range: "<1.0.1"
+    fixed_versions:
+    - "v1.0.1"
+    urls:
+    - "https://github.com/amphp/http/pull/4"
+    cwe_ids:
+    - "CWE-1035"
+    - "CWE-937"
+    identifiers:
+    - "GMS-2018-26"
+    """
+    with open(file, "r") as f:
+        gitlab_advisory = yaml.safe_load(f)
+    if not isinstance(gitlab_advisory, dict):
+        logger.error(f"parse_yaml_file: yaml_file is not of type `dict`: {gitlab_advisory!r}")
+        return
+
+    # refer to schema here https://gitlab.com/gitlab-org/advisories-community/-/blob/main/ci/schema/schema.json
+    aliases = gitlab_advisory.get("identifiers")
+    summary = ". ".join([gitlab_advisory.get("title"), gitlab_advisory.get("description")])
+    urls = gitlab_advisory.get("urls")
+    references = [Reference.from_url(u) for u in urls]
+    date_published = dateparser.parse(gitlab_advisory.get("pubdate"))
+    date_published = pytz.utc.localize(date_published)
+    fixed_versions = gitlab_advisory.get("fixed_versions")
+    affected_version_range = None
+    affected_range = gitlab_advisory.get("affected_range")
+    purl: PackageURL = get_purl(gitlab_advisory.get("package_slug"))
+    if not purl:
+        logger.error(f"parse_yaml_file: purl is not valid: {file!r}")
+        return AdvisoryData(
+            aliases=aliases,
+            summary=summary,
+            references=references,
+            date_published=date_published,
+        )
+    vrc: VersionRange = RANGE_CLASS_BY_SCHEMES[purl.type]
```

Comment:
```suggestion
    if affected_range: 
        vrc: VersionRange = RANGE_CLASS_BY_SCHEMES[purl.type]
        ......
```

e.g., indent all the block